database/rrdcalctemplate.c

```diff
@@ -4,6 +4,32 @@
 #include "rrd.h"
 
 // ----------------------------------------------------------------------------
+
+static int rrdcalctemplate_has_label(RRDCALCTEMPLATE *rt,  RRDHOST *host) {
+    errno = 0;
+    struct label *move = host->labels;
+    if(move && rt->labels) {
+        netdata_rwlock_wrlock(&host->labels_rwlock);
```

Comment:
This should be the reader lock.

Reply:
I agree 100% with this and the necessary change to the next lock, I will write an answer here that is also applied for all other comments that you did, this is a [WIP] yet, so sometimes I only copy and paste the code to test whether the main goal is reached, but before to remove the [WIP] I do a revision of everything.
Thank you very much to save me time, I already brought all the changes that you requested, you save me time! :)